tests/test_llm.py

```diff
@@ -32,10 +32,30 @@ def test_llm_sync():
     expected_result = {"name": "Carla", "age": "25"}
     assert result == expected_result
 
+    # test structured failure
+
+    runtime.api_key = "fake_api_key"
+
+    result = runtime.record_to_record(
+        record={"input_name": "Carla", "input_age": 25},
+        input_template="My name is {input_name} and I am {input_age} years old.",
+        instructions_template="",
+        output_template="name: {name}, age: {age}",
+    )
+
+    expected_result = {
+        '_adala_error': True,
+        '_adala_message': 'AuthenticationError',
+        '_adala_details': "litellm.AuthenticationError: AuthenticationError: OpenAIException - Error code: 401 - {'error': {'message': 'Incorrect API key provided: fake_api_key. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}"
```

Comment:
maybe just worth asserting that something like "AuthenticationError" is in the details string? Would hate for this to start failing for minor format changes from litellm

Reply:
gonna hold off on this for now bc displaying the full error in the test is useful for docs, but def open to not asserting on it